libs/experimental/langchain_experimental/graph_transformers/llm.py

```diff
@@ -320,19 +475,39 @@ def __init__(
         prompt: ChatPromptTemplate = default_prompt,
         strict_mode: bool = True,
     ) -> None:
-        if not hasattr(llm, "with_structured_output"):
-            raise ValueError(
-                "The specified LLM does not support the 'with_structured_output'. "
-                "Please ensure you are using an LLM that supports this feature."
-            )
         self.allowed_nodes = allowed_nodes
         self.allowed_relationships = allowed_relationships
         self.strict_mode = strict_mode
+        self._function_call = True
+        # Check if the LLM really supports structured output
+        try:
+            llm.with_structured_output(_Graph)
+        except NotImplementedError:
+            self._function_call = False
+        if not self._function_call:
+            try:
+                import json_repair
 
-        # Define chain
-        schema = create_simple_model(allowed_nodes, allowed_relationships)
-        structured_llm = llm.with_structured_output(schema, include_raw=True)
-        self.chain = prompt | structured_llm
+                self.json_repair = json_repair
+            except ImportError:
+                raise ImportError(
+                    "Could not import json_repair python package. "
+                    "Please install it with `pip install json-repair`."
+                )
+            self._function_call = False
+            # Allow passing prompts from users
+            if not prompt == default_prompt:
```

Comment:
nit: can we give `prompt` a default value of None so we can more rigorously check if user has passed in a value or not?

Reply:
Fixed